src/python/pants/backend/tools/trufflehog/subsystem.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+"""Downloads the trufflehog binary."""
+
+from __future__ import annotations
+
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.core.util_rules.external_tool import ExternalTool
+from pants.engine.platform import Platform
+from pants.option.option_types import BoolOption, SkipOption, StrListOption
+from pants.util.strutil import softwrap
+
+
+class Trufflehog(ExternalTool):
+    """A tool for finding secrets in code."""
+
+    options_scope = "trufflehog"
+    name = "Trufflehog"
+    default_version = "v3.34.0"
+    help = "Trufflehog secrets scanning"
+    skip = SkipOption("lint")
+    default_known_versions = [
+        f"{default_version}|macos_arm64|19e10e34e95d797cbb924b342b873caa9e71296f9bead28b390d96981f47fbb0|26743363",
```

Comment:
We actually *do* want to hardcode the version here instead of interpolating it, because `default_version` is a moving target. When we add a new version of Trufflehog here, these rows must still exist to support the earlier version, but now the new version may become the default, at which point the new rows should have the new default version. So to avoid confusion, these should all be hardcoded.

Reply:
Ah I see what you mean, hardcoded `default_version` as well